modules/core/02-client/migrations/v7/store.go

```diff
@@ -0,0 +1,215 @@
+func handleTendermintMigration(ctx sdk.Context, store sdk.KVStore, cdc codec.BinaryCodec) error {
+	clients, err := collectClients(ctx, store, exported.Tendermint)
+	if err != nil {
+		return err
+	}
+
+	if len(clients) > 1 {
+		return sdkerrors.Wrap(sdkerrors.ErrLogic, "too many tendermint clients collected")
+	}
+
+	clientID := clients[0]
+
+	clientPrefix := []byte(fmt.Sprintf("%s/%s/", host.KeyClientStorePrefix, clientID))
+	clientStore := prefix.NewStore(store, clientPrefix)
+
+	bz := clientStore.Get(host.ClientStateKey())
+	if bz == nil {
+		return clienttypes.ErrClientNotFound
+	}
+
+	var clientState exported.ClientState
+	if err := cdc.UnmarshalInterface(bz, &clientState); err != nil {
+		return sdkerrors.Wrap(err, "failed to unmarshal client state bytes into tendermint client state")
+	}
+
+	_, ok := clientState.(*ibctm.ClientState)
+	if !ok {
+		return sdkerrors.Wrap(clienttypes.ErrInvalidClient, "client state is not tendermint even though client id contains 07-tendermint")
+	}
+
+	return nil
+}
+
+// handleLocalhostMigration removes all client and consensus states associated with the localhost client type.
+func handleLocalhostMigration(ctx sdk.Context, store sdk.KVStore, cdc codec.BinaryCodec) error {
+	clients, err := collectClients(ctx, store, Localhost)
+	if err != nil {
+		return err
+	}
+
+	for _, clientID := range clients {
+		clientPrefix := []byte(fmt.Sprintf("%s/%s/", host.KeyClientStorePrefix, clientID))
+		clientStore := prefix.NewStore(store, clientPrefix)
+
+		// delete the client state
+		clientStore.Delete(host.ClientStateKey())
+
+		pruneClientConsensusStates(clientStore)
+	}
+
+	return nil
+}
+
+// collectClients will iterate over the provided client type prefix in the client store
+// and return a list of clientIDs associated with the client type. This is necessary to
+// avoid state corruption as modifying state during iteration is unsafe. A special case
+// for tendermint clients is included as only one tendermint clientID is required for
+// v7 migrations.
+func collectClients(ctx sdk.Context, store sdk.KVStore, clientType string) (clients []string, err error) {
+	clientPrefix := []byte(fmt.Sprintf("%s/%s", host.KeyClientStorePrefix, clientType))
+	iterator := sdk.KVStorePrefixIterator(store, clientPrefix)
+
+	defer iterator.Close()
+	for ; iterator.Valid(); iterator.Next() {
+		path := string(iterator.Key())
+		if !strings.Contains(path, host.KeyClientState) {
+			// skip non client state keys
+			continue
+		}
+
+		clientID := host.MustParseClientStatePath(path)
+		clients = append(clients, clientID)
+
+		// optimization: exist after a single tendermint client iteration
+		if strings.Contains(clientID, exported.Tendermint) {
+			return clients, nil
+		}
+	}
+
+	return clients, nil
+}
+
+// pruneClientConsensusStates removes all client consensus states from the associated
+// client store.
+func pruneClientConsensusStates(clientStore sdk.KVStore) {
```

Comment:
Should we consider renaming this function? Pruning suggests that something is kept/leftover, whereas this is effectively removing the entire localhost consensus state history right?